### PR TITLE
SFT-1371: Introduce Scan QR flow class

### PR DESCRIPTION
**Pull request by jeandudey**

### CODE REVIEW

File: extmod/foundation-rust/src/ur/decoder.rs [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+        Err(e) => {
+            *error = e;
+            return false;
+        }
+    };
+
+    true
+}
+
+/// Returns `true` if the string is an uniform resource, `false` otherwise.
```
Comment (FoundationKen) [inline extmod/foundation-rust/src/ur/decoder.rs:199]: "a uniform"

File: extmod/foundation-rust/src/ur/registry.rs [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+    pub fn ur_type(&self) -> &'static str {
+        match self {
+            Value::Standard(value) => value.ur_type(),
+            Value::PassportRequest(_) => "crypto-request",
+            Value::PassportResponse(_) => "crypto-response",
+        }
+    }
+}
+
+/// An uniform resource.
```
Comment (FoundationKen) [inline extmod/foundation-rust/src/ur/registry.rs:69]: "/// A uniform resource."

**Review by FoundationKen — APPROVED** [review]

**Comment by FoundationKen** [comment]
@jeandudey Reminder that this has conflicts to resolve.

**Comment by jeandudey** [comment]
> @jeandudey Reminder that this has conflicts to resolve.

@FoundationKen dev-v2.1.0 needs to be rebased first.

**Review by FoundationKen — DISMISSED** [review]

**Review by mjg-foundation — APPROVED** [review]

**Pull request merged**